### PR TITLE
Add `TIME_FORMAT()` Function To SQL Plugin

### PR DESCRIPTION
**Pull request by GabeFernandez310**
Signed-off-by: GabeFernandez310 <Gabriel.Fernandez@improving.com>

### Description
Adds the `time_format` function to the SQL plugin. The function takes a `DATE`/`DATETIME`/`TIME`/`TIMESTAMP`/`STRING` in a valid format, and a formatting string (e.g., `"%h %p"`) and outputs a formatted string. If a `DATE` is passed in, it treats the argument as if it is taken at midnight (i.e., `00:00:00`).

Example:
```
opensearchsql> SELECT time_format("1998-01-31 13:14:15.012345", "%f %H %h %I %i %p %r %S %s %T");
fetched rows / total rows = 1/1
+------------------------------------------------------------------------------+
| time_format("1998-01-31 13:14:15.012345", "%f %H %h %I %i %p %r %S %s %T")   |
|------------------------------------------------------------------------------|
| 012345 13 01 01 14 PM 01:14:15 PM 15 15 13:14:15                             |
+------------------------------------------------------------------------------+
```
 
### Issues Resolved
#722
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
  - [X] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: integ-test/src/test/java/org/opensearch/sql/ppl/DateTimeFunctionIT.java [hunk 1]
```diff
@@ -636,7 +636,7 @@ public void testDateFormat() throws IOException {
     String timestamp = "1998-01-31 13:14:15.012345";
     String timestampFormat = "%a %b %c %D %d %e %f %H %h %I %i %j %k %l %M "
         + "%m %p %r %S %s %T %% %P";
-    String timestampFormatted = "Sat Jan 01 31st 31 31 12345 13 01 01 14 031 13 1 "
+    String timestampFormatted = "Sat Jan 01 31st 31 31 012345 13 01 01 14 031 13 1 "
```
Comment (penghuo) [inline integ-test/src/test/java/org/opensearch/sql/ppl/DateTimeFunctionIT.java:639]: what is the change? 12345 -> 012345

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 2]
```diff
@@ -846,6 +847,12 @@ private DefaultFunctionResolver date_format() {
             STRING, DATE, STRING),
         impl(nullMissingHandling(DateTimeFormatterUtil::getFormattedDate),
             STRING, DATETIME, STRING),
+        implWithProperties(
```
Comment (penghuo) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:850]: update javadoc also.

File: integ-test/src/test/java/org/opensearch/sql/ppl/DateTimeFunctionIT.java (hunk 1 above)
Comment (GabeFernandez310) [reply integ-test/src/test/java/org/opensearch/sql/ppl/DateTimeFunctionIT.java:639]: For `DATE_FORMAT` and `TIME_FORMAT` functions, microseconds should be padded with 0s on the left side according to both [MySQL](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-format) and [our documentation](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#date-format) (6 digits of precision should always be returned when using `%f`). This was fixed and the test was changed.

**Review by penghuo — DISMISSED** [review]

**Review by dai-chen — DISMISSED** [review]

**Review by Yury-Fridlyand — DISMISSED** [review]

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java [hunk 3]
```diff
@@ -61,7 +66,7 @@ interface DateTimeFormatHandler {
       .put("%D", (date) -> // %w - Day of month with English suffix
           String.format("'%d%s'", date.getDayOfMonth(), getSuffix(date.getDayOfMonth())))
       .put("%f", (date) -> // %f - Microseconds
-          String.format("'%d'", (date.getNano() / 1000)))
+          String.format(NANO_SEC_FORMAT, (date.getNano() / 1000)))
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java:69]: Love it - but I feel like we should use a constant for all of the surrounding format strings.

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java [hunk 4]
```diff
@@ -1952,6 +1989,220 @@ void testDateFormat(DateFormatTester dft) {
...
+        Arguments.of("%M"),
+        Arguments.of("%W"),
+        Arguments.of("%D"),
+        Arguments.of("%w"),
+        Arguments.of("%U"),
+        Arguments.of("%u"),
+        Arguments.of("%V"),
+        Arguments.of("%v"),
+        Arguments.of("%X"),
+        Arguments.of("%x")
```
Comment (acarbonetto) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:2161]: you could also test formats without `%` or with extra characters like `%%h` or `%hh` or wrong order or extra spaces like `h%` or ` %h`

**Review by acarbonetto — APPROVED** [review]
minor requests

**Review by MaxKsyunz — APPROVED** [review]
LGTM @GabeFernandez310.

**Review by dai-chen — APPROVED** [review]

**Pull request merged**